astropy/io/fits/hdu/compressed/header.py

```diff
@@ -739,18 +562,18 @@ def _image_header_to_bintable_header_and_coldefs(
     # value.
     for idx in itertools.count(1):
         zname = "ZNAME" + str(idx)
-        if zname not in bintable_header:
+        if zname not in bintable.header:
             break
         zval = "ZVAL" + str(idx)
-        if bintable_header[zname] == "NOISEBIT":
+        if bintable.header[zname] == "NOISEBIT":
             if quantize_level is None:
-                quantize_level = bintable_header[zval]
-        if bintable_header[zname] == "SCALE   ":
+                quantize_level = bintable.header[zval]
+        if bintable.header[zname] == "SCALE   ":
             if hcomp_scale is None:
-                hcomp_scale = bintable_header[zval]
-        if bintable_header[zname] == "SMOOTH  ":
+                hcomp_scale = bintable.header[zval]
+        if bintable.header[zname] == "SMOOTH  ":
             if hcomp_smooth is None:
-                hcomp_smooth = bintable_header[zval]
+                hcomp_smooth = bintable.header[zval]
 
     if quantize_level is None:
```

Comment:
This again seems duplicated from the base class. Will these values ever be `None`?

Reply:
Fixed